Sources/joyfill-components-swift/Components/Table/CollectionView/CollectionViewTable.swift

```diff
@@ -112,56 +114,162 @@ extension CollectionViewTable: UICollectionViewDelegate, UICollectionViewDataSou
+        }
+    }
+    
+    // Function to set text value after matching column id
+    func setCellTextValue(cell: CollectionViewCell, indexPath: IndexPath) {
+        let cellData = valueData[indexPath.section-1].cells ?? [:]
+        if let matchData = cellData.first(where: {$0.key == tableColumnOrderId[indexPath.row]}) {
+            cell.cellTextView.text = matchData.value
+        } else {
+            cell.cellTextView.text = ""
+        }
+    }
+    
+    // Function to handle cells according to displayMode
+    func viewTableDispalyMode(at cell: CollectionViewCell) {
         switch tableDisplayMode {
         case "readonly":
-            cell.cellTextField.isUserInteractionEnabled = false
+            cell.cellTextView.isUserInteractionEnabled = false
         default:
-            cell.cellTextField.isUserInteractionEnabled = true
+            cell.cellTextView.isUserInteractionEnabled = true
         }
-        
-        cell.cellTextField.borderWidth = 0
-        cell.cellTextField.containerRadius = 0
-        cell.cellTextField.text = "Text 1 at 2"
-        cell.cellTextField.textAlignment = .center
-        cell.cellTextField.selectedBorderWidth = 1
-        cell.cellTextField.selectedCornerRadius = 1
-        cell.cellTextField.deselectedBorderWidth = 0
-        cell.cellTextField.deselectedCornerRadius = 0
-        cell.cellTextField.deselectedBorderColor = .clear
-        cell.cellTextField.font = UIFont.systemFont(ofSize: 12)
-        cell.cellTextField.setOutlineLineWidth(0, for: .normal)
-        cell.cellTextField.setOutlineLineWidth(0, for: .editing)
-        cell.cellTextField.selectedBorderColor = UIColor(hexString: "#1F6BFF") ?? .systemBlue
-        
-        cell.contentView.subviews.forEach { $0.removeFromSuperview() }
-        cell.setupSeparator()
-        cell.setupTextField()
     }
     
     // Function to handle collectionView header
-    func collectionViewHeader(cell: CollectionViewCell, indexPath: IndexPath) {
+    func viewTableCollectionViewHeader(cell: CollectionViewCell, indexPath: IndexPath) {
         switch indexPath.section {
         case 0:
             let cellLabel = Label()
             cellLabel.borderWidth = 1
             cellLabel.borderCornerRadius = 0
             cellLabel.textAlignment = .center
-            cellLabel.labelText = tableHeading[indexPath.row]
+            cellLabel.numberOfLines = 1
+            cellLabel.labelText = tableColumnTitle[indexPath.row]
             cellLabel.backgroundColor = UIColor(hexString: "#F3F4F8")
             cellLabel.borderColor = UIColor(hexString: "#E6E7EA") ?? .lightGray
-            cellLabel.frame = CGRect(x: 0, y: 0, width: cellWidth, height: cellHeight)
-            cellLabel.font = UIFont.boldSystemFont(ofSize: 10)
+            
+            if tableColumnOrderId.count == 1 {
+                cellLabel.frame = CGRect(x: 0, y: 0, width: self.frame.width, height: 50.0)
+            } else if tableColumnOrderId.count == 2 {
```

Comment:
Here is a good area for explanations of what these "if else"'s are doing

Reply:
Done